x-pack/plugins/spaces/server/lib/space_schema.test.ts

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { spaceSchema } from './space_schema';
+
+const defaultProperties = {
+  id: 'foo',
+  name: 'foo',
+};
+
+describe('#id', () => {
+  test('is required', () => {
+    expect(() =>
+      spaceSchema.validate({
+        ...defaultProperties,
+        id: undefined,
+      })
+    ).toThrowErrorMatchingInlineSnapshot(
+      `"[id]: expected value of type [string] but got [undefined]"`
+    );
+  });
+
+  test('allows lowercase a-z, 0-9, "_" and "-"', () => {
+    expect(() =>
+      spaceSchema.validate({
+        ...defaultProperties,
+        id: 'abcdefghijklmnopqrstuvwxyz0123456789_-',
+      })
+    ).not.toThrowError();
```

Comment:
__question:__ why not `toThrowErrorMatchingInlineSnapshot` here as well?

Reply:
This is testing that an error is **not** thrown, so I don't think there's benefit to testing against a specific snapshot?

Reply:
Haha, silly me! Sorry, haven't noticed `not` :see_no_evil: